src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3197,6 +3205,9 @@ void EmitLazy(RegexNode node)
                     string backtrack = ReserveName($"LazyLoopBacktrack");
                     MarkLabel(backtrack, emitSemicolon: false);
 
+                    // We're backtracking.  Check the timeout.
+                    EmitTimeoutCheckIfNeeded(writer, rm);
```

Comment:
Oh, nevermind, I guess you now do it when you actually have to backtrack as opposed to each iteration.